lib/galaxy/config/__init__.py

```diff
@@ -260,6 +260,17 @@ def select_one_or_all_paths_from_list():
     def _path_exists(self, path):  # factored out so we can mock it in tests
         return os.path.exists(path)
 
+    def _set_alt_paths(self, option, *alt_paths):
+        # If `option` is not set, check the default. If that path does NOT exist, check *alt_paths.
```

Comment:
``objects`` is the default, but I think if ``files`` is present it should be used even if ``objects`` exists. So I think this is backward, but I mean you could make an argument either way I think.

Reply:
Here's how I thought about this. If `file_path` is not set and we are using the default, there can be only 2 possible values: `objects` and `files` (because anything else would have to be set by the user). Of all the 4 truth table cases (00, 01, 10, 11) only one is interesting: 11 - both exist. In this case, we can safely assume that both have been used (because that directory is created only when a dataset needs to be stored), even though either one could be empty (files deleted by user). If both were used *as defaults*, `files` is the older default, so we'll go with the one that had to be used more recently - `objects`. If either one was used *not* as a default, but set by the user, then it is *no longer set* (by definition), so we treat it as "outdated".

This is after numerous truth tables and head banging.. And I may be wrong. There can be still weird cases (e.g. had previously set to `objects`, then unset and fell back to using the default - `files` - but did not delete the `objects` directory; in this case we'd be pointing to the wrong directory). But for those cases, I think we should simply provide a bit more guidance in the description of `file_path` in `galaxy.yml.sample` and expect the user to have only one directory *if they intend to use it as the default as opposed to setting it*.

Also, I'm purposefully not including `object_store_store_by` in this logic - I think the value of that option may depend on `file_path`, but not the other way round (and even if it does, it should be handled separately).

Reply:
> In this case, we can safely assume that both have been used (

I don't know why, but I have an empty `database/objects` dir. If you flip this around I think it would be OK (as a standalone PR).

Reply:
> I don't know why, but I have an empty `database/objects` dir. If you flip this around I think it would be OK (as a standalone PR).

I don't know why that happened either. Flipping around - do you mean checking `files` *before* `objects`? Also, I may have a simple case that breaks the current/old setup  - posting as a separate comment.

Reply:
Yes. What John is saying, if `files` exists, use it.

Reply:
^Done.